util/key_val_pair_test.go

```diff
@@ -0,0 +1,64 @@
+package util
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type KvPairSuite struct {
+	kvSlice       KeyValuePairSlice
+	kvSliceNested KeyValuePairSlice
+	testMap       map[string]string
+	testMapNested map[string]map[string]string
+	suite.Suite
+}
+
+func TestKvPairSuite(t *testing.T) {
+	suite.Run(t, new(KvPairSuite))
+}
+
+func (s *KvPairSuite) SetupSuite() {
```

Comment:
probably want to reset this in-between tests. 

probably want to test error cases to make sure that: 
- too much testing becomes an error
- not-strings become errors.

Reply:
Added more tests. The setup data doesn't get changed so there isn't a need to reset it